pycbc/waveform/ringdown.py

```diff
@@ -395,34 +504,148 @@ def apply_taper(tau, amp, phi, delta_t, l=2, m=2, inclination=None):
+        A_{lmn} &= A^0_{lmn} \sqrt{2} \sin(\pi/4 +  \Delta \beta_{lmn}).
+   
+    Here, :math:`A^0_{lmn}` is an overall fiducial amplitude (set by the
+    ``amp``) parameter, and
+
+    .. math::
+
+        \Delta \beta_{lmn} &\in [-pi/4, pi/4], \\
+        \Delta \phi_{lmn}  &\in (-pi, pi)
+    
+    are parameters that define the deviation from circular polarization.
+    Circular polarization occurs when both :math:`\Delta \beta_{lmn}` and
+    :math:`\Delta \phi_{lmn}` are zero (this is equivalent to assuming that
+    :math:`h_{l-mn} = (-1)^l h_{lmn}^*`).
+
+    Parameters
+    ----------
+    f_0 : float
+        The central frequency of the damped sinusoid, in Hz.
+    tau : float
+        The damping time, in seconds.
+    amp : float
+        The intrinsic amplitude of the QNM (:math:`A^0_{lmn}` in the above).
+    phi : float
+        The reference phase of the QNM (:math:`\phi_{lmn}`) in the above.
+    delta_t: float
+        The time step (in seconds) to use for the returned waveforms.
+    t_final : float
+        The duration of the waveform (in seconds) to generate.
+    l : int, optional
+        The l index; default is 2.
+    m : int, optional
+        The m index; default is 2.
+    inclination : float, optional
+        The inclination angle (:math:`\theta` in the above). Ignored if
+        ``harmonics='arbitrary'``. Default is 0.
+    azimuthal : float, optional
+        The azimuthal angle (:math:`\varphi` in the above). Ignored if
+        ``harmonics='arbitrary'``. Default is 0.
+    dphi : float, optional
+        The difference in phase between the +m and -m mode
+        (:math:`\Delta \phi_{lmn}` in the above). Default is 0.
+    dbeta : float, optional
+        The angular difference in the amplitudes of the +m and -m mode
+        (:math:`\Delta \beta_{lmn}` in the above). Default is 0. If this and
+        dphi are both 0, will have circularly polarized waves.
+    harmonics : {'spherical', 'spheroidal', 'arbitrary'}, optional
+        Which harmonics to use. See :py:func:`spher_harms for details.
+        Default is spherical.
+    final_spin : float, optional
+        The dimensionless spin of the black hole. Only needed if
+        ``harmonics='spheroidal'``.
+    pol : float, optional
+        Angle to use for +m arbitrary harmonics. Only needed if
+        ``harmonics='arbitrary'``. See :py:func:`spher_harms for details.
+    polnm : float, optional
+        Angle to use for -m arbitrary harmonics. Only needed if
+        ``harmonics='arbitrary'``. See :py:func:`spher_harms for details.
+
+    Returns
+    -------
+    hplus : TimeSeries
+        The plus polarization.
+    hcross : TimeSeries
+        The cross polarization.
+    """
+    # create the times
+    tlen = int(t_final/delta_t) + 1
+    times = numpy.linspace(0, t_final, num=tlen)
+    # evaluate the harmonics
+    if inclination is None:
+        inclination = 0.
+    if azimuthal is None:
+        azimuthal = 0.
```

Comment:
To do: remove if statements on inclination and azimuthal since they both default to 0.

Reply:
Done